widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoContainer.kt

```diff
@@ -0,0 +1,128 @@
+    private var mediaStateId: Int? = null
+    internal var onFlipButtonClickListener: OnClickListener?
+        set(value) {
+            floatingVisitorVideoView.onFlipButtonClickListener = value
+        }
+        get() = floatingVisitorVideoView.onFlipButtonClickListener
+
+    init {
+        LayoutInflater.from(context).inflate(R.layout.visitor_video_layout_view, this)
+        floatingVisitorVideoView = findViewById(R.id.visitor_video_card)
+        floatingVisitorVideoView.contentDescription = stringProvider.getRemoteString(R.string.call_visitor_video_accessibility_label)
+        setVisitorVideoContainerTouchListener()
+        hide()
+    }
+
+    override fun show(state: MediaState?) {
+        if (state != null && isNewMediaState(state)) {
+            floatingVisitorVideoView.showVisitorVideo(
+                state.video.createVideoView(context.asActivity())
+            )
+        }
+        visibility = VISIBLE
+    }
+
+    override fun showFlipCameraButton(flipButtonState: FloatingVisitorVideoView.FlipButtonState) {
+        floatingVisitorVideoView.showFlipCameraButton(flipButtonState)
+    }
+
+    override fun hide() {
+        floatingVisitorVideoView.hideVisitorVideo()
+        visibility = GONE
+    }
+
+    override fun onResume() {
+        floatingVisitorVideoView.onResume()
+    }
+
+    override fun onPause() {
+        floatingVisitorVideoView.onPause()
+    }
+
+    override fun showOnHold() {
+        floatingVisitorVideoView.showOnHold()
+    }
+
+    override fun hideOnHold() {
+        floatingVisitorVideoView.hideOnHold()
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    private fun setVisitorVideoContainerTouchListener() {
+        floatingVisitorVideoView.setOnTouchListener(
+            ViewHelpers.OnTouchListener(
+                { viewLocation },
+                { newPositionX: Float, newPositionY: Float ->
+                    onViewDragged(
+                        newPositionX,
+                        newPositionY
+                    )
+                }
+            ) { _: View? -> }
+        )
+    }
+
+    private val viewLocation: Pair<Int?, Int?>
+        get() = Pair(
+            java.lang.Float.valueOf(floatingVisitorVideoView.x).toInt(),
+            java.lang.Float.valueOf(floatingVisitorVideoView.y).toInt()
+        )
+
+    private fun onViewDragged(positionX: Float, positionY: Float) {
+        var newPositionX = positionX
+        var newPositionY = positionY
+        if (newPositionX < 0) {
+            newPositionX = 0f
+        }
+        if (newPositionY < 0) {
+            newPositionY = 0f
+        }
+        if (newPositionX > width - floatingVisitorVideoView.width) {
+            newPositionX = (width - floatingVisitorVideoView.width).toFloat()
+        }
+        if (newPositionY > height - floatingVisitorVideoView.height) {
+            newPositionY = (height - floatingVisitorVideoView.height).toFloat()
+        }
```

Comment:
It might be good to add comments - what do these conditions mean and why position has to be replaced.

Reply:
Honestly, I don't know what it is. I converted this class from Java. I believe we need it to avoid moving the visitor video container out of the screen.